src/content/1.7/development/components/grid-component.md

```diff
@@ -7,229 +7,404 @@ weight: 2
-                ->setAssociatedColumn('foo')
             )
-        ;
-    }
-
-    // empty collection by default
-    protected function getGridActions()
-    {
-        return (new GridActionCollection())
-            ->add((new SimpleGridAction('bazAction'))
-                ->setName($this->trans('Baz action', [], 'Admin.Advparameters.Feature'))
-                ->setIcon('bar')
+            ->add((new DataColumn('name'))
+                ->setName($this->trans('Name', [], 'Admin.Advparameters.Feature'))
+                ->setOptions([
+                    'field' => 'name',
+                ])
             )
         ;
     }
 }
 ```
 
-## Grid Data Factory
+We have just created a basic Grid Definition factory in which we defined our Grid's id `products`, translatable
+name `Products` and 3 data columns.
+
+{{% notice note %}}
+It is _recommended_ to keep your name translatable. To make that easy
+`AbstractGridDefinitionFactory` provides access to translator via `trans()` method.
+{{% /notice %}}
+
+And finally register your Grid definition factory as a service.
+
+```yaml
+prestashop.core.grid.definition.factory.products_definition_factory:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Definition\Factory\ProductGridDefinitionFactory'
+    parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
+    public: true
+``` 
+
+Most of the time you wont be creating Grid Definition by yourself but delegating this task to other services.
+But in case you need to create Grid Definition by hand, here's how you can do that.
+
+```php
+$productsGridDefinitionFactory = $container->get('prestashop.core.grid.definition.factory.products_definition_factory');
+$productsGridDefinition = $productsGridDefinitionFactory->getDefinition();
 
-As you can imagine, the responsibility of the Grid Data Factory is to retrieve the Grid data, based on the Grid Definition and the Grid Query Builder.
+// you can access all information thats was defined
+$productsGridDefinition->getColumns(); // collection of defined columns
+$productsGridDefinition->getName(); // "Products"
+$productsGridDefinition->getId(); // "products"
+``` 
 
-The only method available of `GridDataFactoryInterface` is `getData` which returns an instance of `GridData`.
-A GridData is an **immutable object** used to store and retrieve the grid data, so if you want to alter this data, you must do it in Grid Data Factory *before* the GridData creation.
+## Search Criteria
 
-You can rely on the implementation provided in the component `GridDataFactory` or create your owns if you have specific needs. 
+In Grid component Search Criteria is used for Grid's data sorting, paginating & filtering. Search Criteria
+can be loaded from database, URL query or anywhere else. 
 
-## Grid Query Builder
+{{% notice info %}}
+Grid component itself does not manage Search Criteria but instead it provides interface for it.
+In PrestaShop [Filters component](#) is used to resolve Search Criteria for Grid.
```

Comment:
I need time to document it, would you mind to create an issue so I won't forget?

Reply:
Yeah I noticed this part was missing, as well as all the "Learn more" links
shouldn't we add "Coming soon" message? Are these pages supposed to be separate pages or sub sections of this one?
I wrote a little section about this in my PR which could be integrated until the full page is written
https://github.com/PrestaShop/docs/pull/134/files#diff-a04004808f283fab9e54e5fb0b03e318R238